gemrb/core/Video/Video.cpp

```diff
@@ -169,6 +171,8 @@ void Video::SetScreenClip(const Region* clip)
 
 bool Video::ToggleFullscreenMode()
 {
+	auto& vars = core->GetDictionary();
+	vars.Set("Full Screen", !fullscreen);
```

Comment:
This should be done in the culprit, WindowManager::HotKey, which will also avoid tangling Video with Interface.

Reply:
Sorry, I can't get it to work.
Tried this and variants.
```diff
 bool WindowManager::HotKey(const Event& event) const
 {
        if (event.type == Event::KeyDown && event.keyboard.repeats == 1) {
+               auto& vars = core->GetDictionary();
+               bool fullscreen;
                switch (event.keyboard.keycode) {
                        case 'f':
                                video->ToggleFullscreenMode();
+                               fullscreen = (bool) vars.Get("Full Screen");
+                               vars.Set("Full Screen", !fullscreen);
                                return true;
                        case GEM_GRAB:
                                video->ToggleGrabInput();
```

Reply:
You're using the wrong `Get` — the one without a fallback returns a pointer, so that's why `fullscreen` remains always true.